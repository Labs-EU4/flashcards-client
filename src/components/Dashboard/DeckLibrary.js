import React, {useState, useEffect} from "react";
import {getAllDecks} from "../../state/actions/decks/decksActions";
import {connect} from "react-redux";
import styles from "./DeckLibrary.module.css";
import {Link} from "react-router-dom";
import {Card, Icon} from "antd";

export function DeckLibrary() {
  const [decks, setDecks] = useState([{}]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    console.log("hi");

    getAllDecks()();
  }, []);

  return (
    <div className={styles.deckLibraryContainer} data-testid="LibraryContainer">
      <h1>This is deck library</h1>
      {decks.map(deck => {
        return (
          <Card
            role="deck"
            data-testid={`Card${deck.deck_id}`}
            key={deck.deck_id}
            actions={[
              <Link to={`/decks/${deck.deck_id}`}>
                <Icon style={{color: "rgba(14,12,12,.60)"}} type="edit" key="edit" />
              </Link>,
              <Icon type="delete" style={{color: "rgba(14,12,12,.60)"}} key="delete" />,
            ]}
            loading={loading}
            className={styles.deckCard}
          >
            {deck.deck_name}
          </Card>
        );
      })}
    </div>
  );
}

export default connect(state => state, {getAllDecks})(DeckLibrary);
