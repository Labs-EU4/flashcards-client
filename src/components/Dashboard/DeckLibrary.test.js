import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import DeckLibrary from "./DeckLibrary";
import {MemoryRouter} from "react-router-dom";

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(
    <MemoryRouter>
      <DeckLibrary />
    </MemoryRouter>
  );
});

const LibraryContainer = () => {
  return wrapper.queryByTestId("LibraryContainer");
};

const DeckCard = () => {
  return wrapper.queryAllByRole("deck");
};

it("renders without crashing", () => {
  expect(wrapper.container).toMatchSnapshot();
});

it("renders correct elements before response is received", () => {
  expect(LibraryContainer()).toBeInTheDocument();
  expect(DeckCard()[0]).not.toBeDefined();
  expect(DeckCard()).toHaveLength(0);
});

// jest.mock("../../helpers/axiosWithAuth", () => {
//   return () =>
//     Promise.resolve({
//       data: {
//         data: [
//           {
//             name: "Statistical Learning",
//             user_id: 1,
//             public: true,
//           },
//           {
//             name: "Economics",
//             user_id: 1,
//             public: true,
//           },
//           {
//             name: "Maths",
//             user_id: 1,
//             public: true,
//           },
//         ],
//       },
//     });
// });

// it("renders response data correctly", async () => {
//   await axiosWithAuth();
//   expect(axiosWithAuth).toHaveBeenCalledTimes(1);
// });

afterEach(rtl.cleanup);
