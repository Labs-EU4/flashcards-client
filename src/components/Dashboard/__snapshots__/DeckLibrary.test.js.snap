// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="deckLibraryContainer"
    data-testid="LibraryContainer"
  />
</div>
`;
